src/dotnet/commands/dotnet-run/RunCommand.cs

```diff
@@ -38,6 +38,34 @@ public int Start()
                 .ExitCode;
         }
 
+        public RunCommand(string configuration,
+            string framework,
+            bool noBuild,
+            string project,
+            IReadOnlyCollection<string> args)
+        {
+            Configuration = configuration;
+            Framework = framework;
+            NoBuild = noBuild;
+            Project = project;
+            Args = args;
+        }
+
+        public RunCommand MakeNewWithReplaced(string configuration = null,
```

Comment:
I am basically want to do this https://github.com/dotnet/csharplang/issues/322
It is still a WIP for C# language